### PR TITLE
ICU-20235 Undefine OVERFLOW in case Microsoft math.h has defined it.

### PR DESCRIPTION
**Pull request by roubert**

### CODE REVIEW

File: icu4c/source/common/umutablecptrie.cpp [hunk 1]
```diff
@@ -18,6 +18,11 @@
 #include "uassert.h"
 #include "ucptrie_impl.h"
 
+// ICU-20235 In case Microsoft math.h has defined this, undefine it.
+#ifdef OVERFLOW
```
Comment (markusicu) [inline icu4c/source/common/umutablecptrie.cpp]: I don't remember -- does undef complain if the macro was not defined?

**Comment by markusicu** [comment]
Looks like AppVeyor gets killed at an arbitrary time?

File: icu4c/source/common/umutablecptrie.cpp (hunk 1 above)
Comment (jefgen) [reply icu4c/source/common/umutablecptrie.cpp]: Not sure off-hand about other compilers, but with MSVC it doesn't complain:
> You can also apply the #undef directive to an identifier that has no previous definition. This ensures that the identifier is undefined.

**Comment by jefgen** [comment]
I'm not sure what happened to the AppVeyor build, though it seems you can manually request a re-build, (which I did for this PR).

File: icu4c/source/common/umutablecptrie.cpp (hunk 1 above)
Comment (roubert) [reply icu4c/source/common/umutablecptrie.cpp]: I don't know of any compiler that actually requires the #ifdef/#undef incantation, but it's traditional as a way to express that a symbol might in some cases or in some environments be defined, and it's used in a number of places throughout the ICU4C code base.

**Pull request merged**